components/gitpod-protocol/go/gitpod-service.go

```diff
@@ -218,6 +218,8 @@ type ConnectToServerOpts struct {
 	Log                 *logrus.Entry
 	ReconnectionHandler func()
 	CloseHandler        func(error)
+	ComponentName       string
```

Comment:
i think it is better to be transparent, i.e. `UserAgent` and client should pass `gitpod/supervisor`